azext_edge/edge/providers/check/base.py

```diff
@@ -73,7 +75,43 @@ def check_post_deployment(
         for resource, evaluate_func in evaluate_funcs.items():
             if (resource == CORE_SERVICE_RUNTIME_RESOURCE) or\
                     (resource.value in lowercase_api_resources and check_resources[resource]):
-                result["postDeployment"].append(evaluate_func(detail_level=detail_level, as_list=as_list))
+                result["postDeployment"].append(evaluate_func(detail_level=detail_level, as_list=as_list, resource_name=resource_name))
+
+
+def filter_resources_by_name(
+    resources: List[dict],
+    resource_name: str,
+) -> List[dict]:
+    # decide if to use resource["metadata"]["name"] or resource.metadata.name
+    def get_resource_name(resource: dict) -> str:
+        # get lowercase name from resource
+        if isinstance(resource, dict):
+            return resource.get("metadata", {}).get("name", "").lower()
+        return resource.metadata.name.lower()
+
+    if resource_name:
+        resource_name = resource_name.lower()
+        # check resource for wildcarded name
+        if "*" in resource_name:
+            resource_name = resource_name.replace("*", ".*")
+            # add beginning and end of string regex
+            resource_name = f"^{resource_name}$"
+            resources = [resource for resource in resources if re.match(resource_name, get_resource_name(resource))]
+        else:
+            resources = [resource for resource in resources if resource_name == get_resource_name(resource)]
```

Comment:
I did a bit of testing, and I think we can improve performance here for wildcard searches with a small tweak.
We can use `re.compile()` to speed up multiple regex matches (so you don't need to compile the regex again for each item in the list):
```suggestion
        if "*" in resource_name:
            regex = re.compile(f"^{resource_name.replace('*', '.*')}$")
            resources = [resource for resource in resources if regex.match(get_resource_name(resource))]
        else:
            resources = [resource for resource in resources if resource_name == get_resource_name(resource)]
```